crypto/evp/evp_fetch.c

```diff
@@ -68,16 +69,23 @@ static OSSL_METHOD_STORE *get_default_method_store(OPENSSL_CTX *libctx)
 }
 
 static void *get_method_from_store(OPENSSL_CTX *libctx, void *store,
-                                   const char *propquery, void *data)
+                                   const char *name, const char *propquery,
+                                   void *data)
 {
     struct method_data_st *methdata = data;
     void *method = NULL;
+    OSSL_NAMEMAP *namemap;
+    int nid;
 
     if (store == NULL
         && (store = get_default_method_store(libctx)) == NULL)
         return NULL;
 
-    (void)ossl_method_store_fetch(store, methdata->nid, propquery, &method);
+    if ((namemap = ossl_namemap_stored(libctx)) == NULL
+        || (nid = ossl_namemap_add(namemap, name)) == 0)
```

Comment:
While this is a nid in the sense that nid == numeric identifier, it is a different type of nid to where that term is used everywhere else in the code. I think that's confusing so a different name would be a good idea.

Reply:
I can certainly go with `id` if that's preferable.

Reply:
Yes, I think that would be less confusing.

Reply:
Okie